src/main/java/frc/robot/subsystems/ball/feeder/UltrasonicPoller.java

```diff
@@ -10,21 +10,63 @@
 
     Ultrasonic ultrasonicSensor;
 
-    UltrasonicPoller(Ultrasonic sensor, int time) {
+    // Consider renaming, couldn't think of a better one ATM
+    boolean noBalls;
+    long time;
+    double distance;
+    double currentRange;
+    int count;
+
+    long startTime;
+
+    UltrasonicPoller(Ultrasonic sensor, long time, double distance, long rate) {
         // sensor is pretty self explanatory
         // time is how long you want the sensor to not detect an object until true is returned
-        super(25, TimeUnit.MILLISECONDS); // 4Hz seems reasonable
+        // distance is the minimum distance threshold
+        super(rate, TimeUnit.MILLISECONDS); // 4Hz seems reasonable
         this.ultrasonicSensor = sensor;
+        this.time = time;
+        this.distance = distance;
+        this.noBalls = false;
+        this.startTime = 0;
+        this.count = 0; // There has to be a better way to do this
+
+    }
+
+    UltrasonicPoller(Ultrasonic sensor, long time, double distance) {
+        this(sensor, time, distance, 25);
     }
 
     @Override
     public void defineResources(){
         require(ultrasonicSensor);
     }
 
+    public void setTime(long time) {
+        this.time = time;
+    }
+
+    public void setDistance(double distance){
+        this.distance = distance;
+    }
+
+    // Consider renaming, probably want to generalize the name if we ever re-use it
+    public boolean checkUnloaded(){
+        return this.noBalls;
+    }
+
     @Override
     public void task(){
-        // Placholder
-        return;
+        this.currentRange = this.ultrasonicSensor.getRangeMM();
+        if(this.count == 0){ // TODO look into a better way of doing this
```

Comment:
Okay, this is getting a little messy and there are definitely ways to streamline this and keep fewer redundant fields, but I think at this point your best bet is probably going to be to just test this out on real hardware and see how it behaves. That'll help you get a good mental model of what information is available and how you can best model it.

Reply:
Just figured out how I could rewrite it
```
do {
set the initial time
} while the initial time is zero
// rest of the task
```

Reply:
That sounds a lot like

```
if initial time is zero
  set the initial time
```

no?

Reply:
Dude how did I miss that. It's been there in front of me this entire time but I was too blind to see it.

Reply:
Granted - both of these approaches will cause an issue if the system is stopped and then started again (but that issue is insignificant because I doubt that the output of the poller will be checked within the 20ms timeframe that noBalls is an incorrect value), but how likely are we to do that. Now that I think about it, is it even necessary to do this? If noBalls holds an incorrect value when the subsystem is started, it's only for 20ms until the loop runs again, where if it detects a ball it will set noBalls to false. However, that does cause issues where, say the threshold is 3 seconds and there are no balls for two seconds, it will return a true value, even though it should return false.

Reply:
I'm not really following your explanation, but yea, there are still edge cases you might need to iron out. I think what you have here is workable enough to test on real hardware and tinker with it from there though, so I'm going to merge this and you can open a new PR with any refinements.

Reply:
That's implying the robot will actually be together in time for me to do any tinkering